pkg/bitlpm/cidr.go

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package bitlpm
+
+import (
+	"math/bits"
+	"net/netip"
+)
+
+// CIDRTrie can hold both IPv4 and IPv6 prefixes
+// at the same time.
+type CIDRTrie[T any] struct {
+	v4 Trie[Key[netip.Prefix], T]
+	v6 Trie[Key[netip.Prefix], T]
+}
+
+// NewCIDRTrie creates a new CIDRTrie[T any].
+func NewCIDRTrie[T any]() *CIDRTrie[T] {
+	return &CIDRTrie[T]{
+		v4: NewTrie[netip.Prefix, T](32),
+		v6: NewTrie[netip.Prefix, T](128),
+	}
+}
+
+// Lookup returns the longest matched value for a given address.
+func (c *CIDRTrie[T]) Lookup(addr netip.Addr) T {
+	tri := c.v4
+	bits := 32
+	if addr.Is6() {
+		tri = c.v6
+		bits = 128
+	}
+	var def T
+	tri.Ancestors(uint(bits), cidrKey(netip.PrefixFrom(addr, bits)), func(_ uint, _ Key[netip.Prefix], v T) bool {
+		def = v
+		return true
+	})
+	return def
+}
+
+// Path calls fn for every entry that contains a given prefix.
```

Comment:
nit: Was there deliberate difference in terminology here for `Path` for the CIDRTrie to make it easier to reason about, or was this intended to match `Ancestors` of the underlying structure, it's just that one was renamed and the other wasn't? Consistency is a nice-to-have unless the inconsistency makes usage clearer. Off the top of my head I wouldn't have thought of the set of possible CIDR prefixes as a "Path" but mostly because networking already defines other meaning for the word "Path".

Reply:
Yeah. I agree. Let's keep naming consistent.